sdk/spring/spring-cloud-azure-autoconfigure/src/test/java/com/azure/spring/cloud/autoconfigure/aad/implementation/webapp/AadOAuth2UserServiceTest.java

```diff
@@ -0,0 +1,209 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+package com.azure.spring.cloud.autoconfigure.aad.implementation.webapp;
+
+import com.azure.spring.cloud.autoconfigure.aad.implementation.constants.AuthorityPrefix;
+import com.azure.spring.cloud.autoconfigure.aad.implementation.graph.GraphClient;
+import com.azure.spring.cloud.autoconfigure.aad.implementation.graph.GroupInformation;
+import com.azure.spring.cloud.autoconfigure.aad.properties.AadAuthenticationProperties;
+import org.junit.jupiter.api.AfterEach;
+import org.junit.jupiter.api.BeforeEach;
+import org.junit.jupiter.api.Test;
+import org.springframework.mock.web.MockHttpServletRequest;
+import org.springframework.security.core.Authentication;
+import org.springframework.security.core.authority.SimpleGrantedAuthority;
+import org.springframework.security.core.context.SecurityContextHolder;
+import org.springframework.security.oauth2.client.oidc.userinfo.OidcUserRequest;
+import org.springframework.security.oauth2.client.registration.ClientRegistration;
+import org.springframework.security.oauth2.core.AuthenticationMethod;
+import org.springframework.security.oauth2.core.AuthorizationGrantType;
+import org.springframework.security.oauth2.core.OAuth2AccessToken;
+import org.springframework.security.oauth2.core.oidc.IdTokenClaimNames;
+import org.springframework.security.oauth2.core.oidc.OidcIdToken;
+import org.springframework.security.oauth2.core.oidc.OidcScopes;
+import org.springframework.security.oauth2.core.oidc.StandardClaimNames;
+import org.springframework.security.oauth2.core.oidc.user.DefaultOidcUser;
+import org.springframework.security.oauth2.core.oidc.user.OidcUser;
+import org.springframework.web.context.request.RequestContextHolder;
+import org.springframework.web.context.request.ServletRequestAttributes;
+
+import javax.servlet.http.HttpSession;
+import java.time.Instant;
+import java.util.HashMap;
+import java.util.Map;
+import java.util.Set;
+import java.util.stream.Collectors;
+import java.util.stream.Stream;
+
+import static org.assertj.core.api.Assertions.assertThat;
+import static org.assertj.core.api.Assertions.assertThatIllegalArgumentException;
+import static org.mockito.ArgumentMatchers.anyString;
+import static org.mockito.Mockito.RETURNS_DEEP_STUBS;
+import static org.mockito.Mockito.mock;
+import static org.mockito.Mockito.when;
+
+/**
+ * Tests for {@link AadOAuth2UserService}.
+ */
+public class AadOAuth2UserServiceTest {
+    private ClientRegistration.Builder clientRegistrationBuilder;
+    private OidcIdToken idToken;
+    private AadOAuth2UserService aadOAuth2UserService;
+    private OAuth2AccessToken accessToken;
+    private Map<String, Object> idTokenClaims = new HashMap<>();
+    private GraphClient graphClient;
+    private AadAuthenticationProperties properties;
+
+    private static final String DEFAULT_OIDC_USER = "defaultOidcUser";
+
+
+    @BeforeEach
+    void setup() {
```

Comment:
Do you think we can make the `private DefaultOidcUser getUser(OidcUserRequest userRequest)` method as package private, so we can test that directly? So we don't need to setup / clean the `SecurityContextHolder.getContext().getAuthentication();` for each case.

Reply:
Good point, it might be ok.